core/src/test/java/kafka/testkit/KafkaClusterTestKit.java

```diff
@@ -121,17 +122,25 @@ public Builder setConfigProp(String key, String value) {
         public KafkaClusterTestKit build() throws Exception {
             Map<Integer, Kip500Controller> controllers = new HashMap<>();
             Map<Integer, Kip500Broker> kip500Brokers = new HashMap<>();
+            Map<Integer, KafkaRaftManager> raftManagers = new HashMap<>();
+            String dummyQuorumVotersString = nodes.controllerNodes().keySet().stream().
```

Comment:
At a high level, I think we should make `0.0.0.0:0` a valid address configuration from the perspective of `KafkaRaftClient`. We are telling the client that initialization of the endpoints will come at a later time and through some external mechanism. I think this is a generally useful capability even outside of testing. For example, we may eventually want to be able to initialize an observer from a `bootstrap.servers` configuration. In this case, discovery of the voter endpoints will come after we have (say) sent a `DescribeQuorum` to the cluster. We may also want to implement some of the more fancy bootstrapping options that etcd has, such as using an external service to facilitate address bootstrapping.

Anyway, my point is we should treat this as less of a hack and more using a supported capability. To that end, I think we should document the behavior and add tests for it (we can do this separately of course).

Reply:
Yea, I think that's a fair point. Delaying the raft voter configuration has other applications as you said and I think we should do this in another PR that can also be then upstreamed